File: pet-clinic-data/src/main/java/com/sachet/sfgpetclinic/model/Vet.java
package com.sachet.sfgpetclinic.model;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "vets")
public class Vet extends Person{

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "vet_specialities", joinColumns = @JoinColumn(name = "vet_id"), inverseJoinColumns = @JoinColumn(name = "speciality_id"))
    Set<Speciality> specialities;

    public Set<Speciality> getSpecialities() {
        return specialities;
    }

    public void setSpecialities(Set<Speciality> specialities) {
        this.specialities = specialities;
    }
}

File: src/main/kotlin/com/sachet/sfgpetclinic/model/Vet.kt
package com.sachet.sfgpetclinic.model

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany

@Entity
class Vet(
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "vet_specialities", joinColumns = [JoinColumn(name = "vet_id")]
    , inverseJoinColumns = [JoinColumn(name = "speciality_id")])
    val specialities: MutableSet<Speciality> = HashSet()
): Person() {
}